### PR TITLE
Stop backing up excluded app APKs

### LINKED ISSUE #594: APKs of apps that don't allow backup do get backed up even if APK backup is off
We may be doing this, so the user can get these apps back. They may not appear in the list of apps to manually install. However, for many apps that don't allow backup, this could lead to significant storage requirements and cause time-outs.

If the reasoning is true, we should store the list of those apps and just add them to the list of apps to restore, even if we don't have data for them.

### PR DESCRIPTION
**Pull request by stevesoltys**
Resolves #594

### CODE REVIEW

**Review by grote — DISMISSED** [review]
Looks good. Need to look into why tests don't pass.

File: app/src/test/java/com/stevesoltys/seedvault/transport/backup/ApkBackupTest.kt [hunk 1]
```diff
@@ -53,6 +54,12 @@ internal class ApkBackupTest : BackupTest() {
         mockkStatic(PackageUtils::class)
     }
 
+    @BeforeEach
+    fun setUp() {
+        every { settingsManager.backupApks() } returns true
+        every { settingsManager.isBackupEnabled(any()) } returns true
+    }
```
Comment (grote) [inline app/src/test/java/com/stevesoltys/seedvault/transport/backup/ApkBackupTest.kt]: For small things like these, I'd prefer to have them explicitly in each test, especially when the test outcome depends on it, e.g. `settingsManager.backupApks()`

Comment (stevesoltys) [reply app/src/test/java/com/stevesoltys/seedvault/transport/backup/ApkBackupTest.kt]: I would argue that establishing a default state for a "happy path" and then having tests which deviate from that path makes the tests easier to understand and reduces duplicate code.

This is subjective though so I've reverted the change.

Comment (grote) [reply app/src/test/java/com/stevesoltys/seedvault/transport/backup/ApkBackupTest.kt]: Thanks!

**Review by grote — APPROVED** [review]

**Pull request merged**